### PR TITLE
Move to version 3.x of the AWS Terraform provider

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description
This pull request upgrades the Terraform code to use to version 3 of the Terraform AWS provider.

It also cleans up and updates the example code.

## 💭 Motivation and Context
Version 3 of the AWS provider has been out for a while. We don't want to get left behind.

The example code was deploying our CyHy bastion AMI to the default VPC.  This PR now deploys the Example AMI (from [`skeleton-packer`](https://github.com/cisagov/skeleton-packer) into a dedicated VPC/subnet (we no longer rely on the existence of a default VPC).
 

## 🧪 Testing
* I successfully ran `terraform apply` and `terraform destroy` with the updated example code and the 3.x AWS provider.
* All automated tests pass.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)
- [x] Provider upgrade

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: examples/basic_usage/main.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  source = "../../"
+  providers = {
+    aws = aws
+  }
+
+  ami_owner_account_id  = var.ami_owner_account_id
+  aws_availability_zone = var.aws_availability_zone
+  aws_region            = var.aws_region
+  subnet_id             = aws_subnet.example.id
+
```
Comment (jsf9k) [inline examples/basic_usage/main.tf]: Please remove this extra line.

File: examples/basic_usage/variables.tf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# ------------------------------------------------------------------------------
+
+# ------------------------------------------------------------------------------
+# Optional parameters
+#
+# These parameters have reasonable defaults.
+# ------------------------------------------------------------------------------
+
+variable "ami_owner_account_id" {
+  description = "The ID of the AWS account that owns the AMI, or \"self\" if the AMI is owned by the same account as the provisioner."
```
Comment (jsf9k) [inline examples/basic_usage/variables.tf:20]: You need a type here.

File: variables.tf [hunk 3]
```diff
@@ -14,6 +14,10 @@ variable "subnet_id" {
 #
 # These parameters have reasonable defaults.
 # ------------------------------------------------------------------------------
+variable "ami_owner_account_id" {
```
Comment (jsf9k) [inline variables.tf:17]: You need a type here.

**Review by jsf9k — CHANGES_REQUESTED** [review]
I don't think the `README.md` file at the root of the repo got updated.

I also requested a few minor changes in my comments.

File: examples/basic_usage/variables.tf (hunk 2 above)
Comment (dav3r) [reply examples/basic_usage/variables.tf:20]: Good eye!  Removed in https://github.com/cisagov/skeleton-tf-module/pull/39/commits/0dd4c0ce78af5acc7eff7ddfdde85209f8ff94ca.

File: examples/basic_usage/main.tf (hunk 1 above)
Comment (dav3r) [reply examples/basic_usage/main.tf]: Good eye!  Removed in https://github.com/cisagov/skeleton-tf-module/pull/39/commits/0dd4c0ce78af5acc7eff7ddfdde85209f8ff94ca.

File: variables.tf (hunk 3 above)
Comment (dav3r) [reply variables.tf:17]: Thanks- fixed in https://github.com/cisagov/skeleton-tf-module/pull/39/commits/1b84540513882933410a958e9205fb11cf8094dc.

File: examples/basic_usage/variables.tf (hunk 2 above)
Comment (dav3r) [reply examples/basic_usage/variables.tf:20]: Thanks- fixed in https://github.com/cisagov/skeleton-tf-module/pull/39/commits/1b84540513882933410a958e9205fb11cf8094dc.

**Comment by jsf9k** [comment]
Still waiting on that `README.md` to get updated via `terraform-docs`, then I will approve.

**Comment by dav3r** [comment]
> I don't think the `README.md` file at the root of the repo got updated.

Thanks- this was taken care of in https://github.com/cisagov/skeleton-tf-module/pull/39/commits/50915d165fd658ec5eddf4e3de4cdb6b1975293b.  At first, I thought you were referring to the README changes from #38, so I pulled in those updates in https://github.com/cisagov/skeleton-tf-module/pull/39/commits/ec979b448b8f8cd6bd0f12571ddbcfe8f897fa1b (in case you're wondering why those changes are now in here).

**Review by jsf9k — APPROVED** [review]
Nice work!

File: versions.tf [hunk 4]
```diff
@@ -6,6 +6,6 @@ terraform {
   # major version currently being used.  This practice will help us
   # avoid unwelcome surprises.
   required_providers {
-    aws = "~> 2.0"
+    aws = "~> 3.0"
```
Comment (felddy) [inline versions.tf:9]: 🎉

**Review by felddy — APPROVED** [review]
Lookin' güd!  👀

**Pull request merged**